### PR TITLE
feat(releases): Replace global date selector with local one

### PR DESCRIPTION
**Pull request by matejminar**
### Before
image

### After
image

This is feature flagged work in progress.

### CODE REVIEW

File: static/app/views/releases/detail/overview/releaseStats.tsx [hunk 1]
```diff
@@ -136,192 +137,200 @@ function ReleaseStats({
...
-              <CrashFreeWrapper>
-                {defined(crashFreeSessions) && (
-                  <div>
-                    <CrashFree
-                      percent={crashFreeSessions}
-                      iconSize="md"
-                      displayOption={DisplayOption.SESSIONS}
-                    />
-                  </div>
+      {!organization.features.includes('release-comparison') && (
```
Comment (matejminar) [inline static/app/views/releases/detail/overview/releaseStats.tsx:140]: This whole diff is just wrapping the block of code with a feature flag condition. No other changes made.

File: static/app/views/releases/detail/overview/index.tsx [hunk 2]
```diff
@@ -323,36 +426,72 @@ class ReleaseOverview extends AsyncView<Props> {
                     onRestore={() => this.handleRestore(project, refetchData)}
                   />
                 )}
+                <Feature features={['release-comparison']}>
+                  {({hasFeature}) =>
+                    hasFeature ? (
+                      <React.Fragment>
```
Comment (priscilawebdev) [inline static/app/views/releases/detail/overview/index.tsx:432]:
```suggestion
                      <Fragment>
```

File: static/app/views/releases/detail/overview/index.tsx [hunk 3]
```diff
@@ -323,36 +426,72 @@ class ReleaseOverview extends AsyncView<Props> {
...
+                                <DateTime date={releaseStart} timeAndDate /> -{' '}
+                                <DateTime date={releaseEnd} timeAndDate />)
+                              </Fragment>
+                            ),
+                            ...DEFAULT_RELATIVE_PERIODS,
+                          }}
+                          defaultPeriod={RELEASE_PERIOD_KEY}
+                        />
+                        {/* TODO(release-comparison): new chart */}
+                      </React.Fragment>
```
Comment (priscilawebdev) [inline static/app/views/releases/detail/overview/index.tsx:453]:
```suggestion
                      </Fragment>
```

File: static/app/views/releases/utils/index.tsx [hunk 4]
```diff
@@ -90,3 +95,58 @@ export function releaseDisplayLabel(displayOption: DisplayOption, count?: number
...
+}
+
+type GetReleaseParams = {
+  location: Location;
+  releaseBounds: ReleaseBounds;
+  defaultStatsPeriod: string;
+  allowEmptyPeriod: boolean;
+};
+
+// these option are here only temporarily while we still support older and newer release details page
```
Comment (priscilawebdev) [inline static/app/views/releases/utils/index.tsx:123]:
```suggestion
// these options are here only temporarily while we still support older and newer release details page
```

**Review by priscilawebdev — APPROVED** [review]
code looks good to me

**Pull request merged**